src/lib/components/ratio/index.svelte

```diff
@@ -8,13 +8,13 @@
   import { bg, fg } from '$lib/stores';
 
   let fgPremultiplied = $derived.by(() => {
-    if ($fg.alpha === 1 || $bg.alpha !== 1) return $fg;
+    if ($fg.alpha === 1 || $bg.alpha !== 1) return null;
     return mix($bg, $fg, $fg.alpha, {
       space: 'srgb',
       premultiplied: false,
     });
   });
-  let ratio = $derived(contrast($bg, fgPremultiplied, 'WCAG21'));
+  let ratio = $derived(contrast($bg, fgPremultiplied ?? $fg, 'WCAG21'));
```

Comment:
This is still a bit mysterious, but I think the root issue is that when `fgPremultiplied` returns `$fg` directly, it apparently doesn't trigger a recalculation of `ratio` even if `$fg` has changed. This change seems to fix it, or we could combine them and this works too:

```ts
  let ratio = $derived.by(() => {
    let fgPremultiplied = $fg;
    if ($fg.alpha !== 1 && $bg.alpha === 1) {
      fgPremultiplied = mix($bg, $fg, $fg.alpha, {
        space: 'srgb',
        premultiplied: false,
      });
    }
    return contrast($bg, fgPremultiplied, 'WCAG21');
  });
```

Reply:
Interesting.

I think I prefer the combined version- it's just a bit cleaner without the `null`, and since we're only really needing to expose `ratio`, and not the `fgPremultiplied` intermediate step.